File: _scratch/run/repeated_substring_pattern/sol.py
class Solution:
    def repeatedSubstringPattern(self, s):
        return s in (s + s)[1:-1]
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+_i] if 1+_i<len(_l) else ""
    _o.append('true' if Solution().repeatedSubstringPattern(_s) else 'false')
print('\n'.join(_o))

File: _scratch/run/repeated_substring_pattern/sol.js
/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function(s) {
    return (s + s).slice(1, -1).indexOf(s) !== -1;
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+_i]||'').replace(/\r$/,'');_o.push(repeatedSubstringPattern(_s)?'true':'false');}
console.log(_o.join('\n'));